raiden/transfer/state.py

```diff
@@ -146,6 +158,195 @@ def __ne__(self, other):
         return not self.__eq__(other)
 
 
+class BalanceProofUnsignedState(State):
+    """ Balance proof from the local node without the signature. """
+
+    __slots__ = (
+        'nonce',
+        'transferred_amount',
+        'locksroot',
+        'channel_address',
+    )
+
+    def __init__(
+            self,
+            nonce: int,
+            transferred_amount: typing.token_amount,
+            locksroot: typing.keccak256,
+            channel_address: typing.address):
+
+        if not isinstance(nonce, int):
+            raise ValueError('nonce must be int')
+
+        if not isinstance(transferred_amount, typing.token_amount):
+            raise ValueError('transferred_amount must be a token_amount instance')
+
+        if not isinstance(locksroot, typing.keccak256):
+            raise ValueError('locksroot must be an keccak256 instance')
+
+        if not isinstance(channel_address, typing.address):
+            raise ValueError('channel_address must be an address instance')
+
+        if nonce <= 0:
+            raise ValueError('nonce cannot be zero or negative')
+
+        if nonce >= 2 ** 64:
+            raise ValueError('nonce is too large')
+
+        if transferred_amount < 0:
+            raise ValueError('transferred_amount cannot be negative')
+
+        if transferred_amount >= 2 ** 256:
+            raise ValueError('transferred_amount is too large')
+
+        if len(locksroot) != 32:
+            raise ValueError('locksroot must have length 32')
+
+        if len(channel_address) != 20:
+            raise ValueError('channel is an invalid address')
+
+        self.nonce = nonce
+        self.transferred_amount = transferred_amount
+        self.locksroot = locksroot
+        self.channel_address = channel_address
+
+    def __str__(self):
+        return (
+            '<'
+            'BalanceProofUnsignedState nonce:{} transferred_amount:{} '
+            'locksroot:{} channel_address:{}'
+            '>'
+        ).format(
+            self.nonce,
+            self.transferred_amount,
+            pex(self.locksroot),
+            pex(self.channel_address),
+        )
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, BalanceProofUnsignedState) and
+            self.nonce == other.nonce and
+            self.transferred_amount == other.transferred_amount and
+            self.locksroot == other.locksroot and
+            self.channel_address == other.channel_address
+        )
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+
+class BalanceProofState2(State):
```

Comment:
Can you rename this class? The `2` at the end is confusing. `SignedBalanceProofState` would be an okay name.